modAionImpl/src/org/aion/zero/impl/sync/statistics/ResponseStatsTracker.java

```diff
@@ -17,45 +19,50 @@
  * @author Beidou Zhang
  */
 public class ResponseStatsTracker {
-
-    // tracks status messages
-    private final ResponseStats status;
-    private final Lock lockStatus;
-
-    // track headers messages
-    private final ResponseStats headers;
-    private final Lock lockHeaders;
-
-    // track bodies messages
-    private final ResponseStats bodies;
-    private final Lock lockBodies;
+    // track status, headers and bodes messages
+    private final EnumMap<RequestType, ResponseStats> responseStats =
+            new EnumMap<>(RequestType.class);
+    private final EnumMap<RequestType, Lock> responseLocks = new EnumMap<>(RequestType.class);
 
     public ResponseStatsTracker(int maxActivePeers) {
-        // instantiate objects for gathering stats
-        this.status = new ResponseStats(maxActivePeers);
-        this.headers = new ResponseStats(maxActivePeers);
-        this.bodies = new ResponseStats(maxActivePeers);
-
-        // instantiate locks
-        this.lockStatus = new ReentrantLock();
-        this.lockHeaders = new ReentrantLock();
-        this.lockBodies = new ReentrantLock();
+        for (RequestType type : RequestType.values()) {
+            // instantiate objects for gathering stats
+            this.responseStats.put(type, new ResponseStats(maxActivePeers));
+            // instantiate locks
+            this.responseLocks.put(type, new ReentrantLock());
+        }
     }
 
     public Map<String, Map<String, Pair<Double, Integer>>> getResponseStats() {
         // acquire lock for all resources
-        lockBodies.lock();
-        lockHeaders.lock();
-        lockStatus.lock();
+        this.responseLocks.get(RequestType.TRIE_DATA).lock();
+        this.responseLocks.get(RequestType.RECEIPTS).lock();
+        this.responseLocks.get(RequestType.BLOCKS).lock();
+        this.responseLocks.get(RequestType.BODIES).lock();
+        this.responseLocks.get(RequestType.HEADERS).lock();
+        this.responseLocks.get(RequestType.STATUS).lock();
 
         try {
-            Map<String, Pair<Double, Integer>> statusStats = this.status.getResponseStatsByPeers();
+            Map<String, Pair<Double, Integer>> statusStats =
```

Comment:
can this be refactored somehow to reduce code repetition?

Reply:
rewrote the method to reduce code repetition